src/directives/column-reorder.js

```diff
@@ -0,0 +1,223 @@
+(function(angular) {
+    "use strict";
+
+    var module = angular.module('table.column-reorder', ['net.enzey.services.events']);
+
+    module.directive('reorderableColumns', function() {
+        var incrementedId = 0;
+        var getId = function() {
+            return 'colId' + incrementedId++;
+        };
+
+        return {
+            controller: function($scope) {
+                var templateMapper = {};
+
+                this.getTemplate = function(rowId) {
+                    return templateMapper[rowId];
+                };
+
+                this.setTemplate = function(rowId, template) {
+                    if (!templateMapper[rowId]) {
+                        templateMapper[rowId] = template;
+                    }
+                };
+
+                this.reorderColumns = function(oldColIndex, newColIndex, offset) {
+
+                    if (oldColIndex === null || newColIndex === null) {
+                        return;
+                    }
+
+                    oldColIndex = +oldColIndex;
+                    newColIndex = +newColIndex;
+
+                    if (oldColIndex < 0 || newColIndex < 0) {
+                        return;
+                    }
+
+                    if (oldColIndex < newColIndex) {
+                        newColIndex--;
+                    }
+
+                    Object.keys(templateMapper).forEach(function(rowId) {
+                        var rowTemplate = templateMapper[rowId];
+                        rowTemplate = angular.element('<tr>' + rowTemplate + '<tr>');
```

Comment:
You are not changing/realigning the col-group's col on drop.  This means if you have a wide Name column and drag it to a small column to swap out it makes the name small.  I think we need to see if there is a size on the col for the old and new columns and replace/transfer/delete the width after move to preserve alignment.